modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -151,12 +163,34 @@ def validate_form_526_claimant_certification
       def validate_form_526_identification
         return if form_attributes['veteranIdentification'].blank?
 
+        addr = form_attributes.dig('veteranIdentification', 'mailingAddress')
+
+        validate_form_526_address_type(addr)
         validate_form_526_current_mailing_address_country
         validate_form_526_current_mailing_address_state
         validate_form_526_current_mailing_address_zip
         validate_form_526_service_number
       end
 
+      def validate_form_526_address_type(addr)
+        validate_military_address(addr) if address_is_military?(addr)
+      end
+
+      def validate_military_address(addr)
+        city = addr['city']
+        state = addr['state']
```

Comment:
Should we move addr into the validation method for consistency? Something like:

```rb
def validate_form_526_identification
  return if form_attributes['veteranIdentification'].blank?

  validate_form_526_military_address
  validate_form_526_current_mailing_address_country
  validate_form_526_current_mailing_address_state
  validate_form_526_current_mailing_address_zip
  validate_form_526_service_number
end

def validate_form_526_military_address
  addr = form_attributes.dig('veteranIdentification', 'mailingAddress')

  return unless address_is_military?(addr)

  city = addr['city']
  state = addr['state']
  # ...
```
?